tools/crm_resource_runtime.c

```diff
@@ -1710,6 +1710,22 @@ cli_resource_execute_from_params(pcmk__output_t *out, const char *rsc_name,
         free(level);
     }
 
+    if (pcmk__str_eq(out->fmt_name, "xml", pcmk__str_none)) {
+        xmlNodePtr node = pcmk__output_xml_create_parent(out, "resource-agent-action",
+                                                         "action", action,
+                                                         "class", rsc_class,
+                                                         "type", rsc_type,
+                                                         NULL);
+
+        if (rsc_name) {
+            crm_xml_add(node, "rsc", rsc_name);
+        }
+
+        if (rsc_prov) {
+            crm_xml_add(node, "provider", rsc_prov);
+        }
+    }
```

Comment:
I really think we need to avoid format-checking to preserve the output model. I'd rather make the whole thing a custom message, and make the common part a function that all messages can call. (Ideally doing the actual work first, rather than intermixing the work and output.)

Reply:
Okay, I'll work on rearranging this.

Reply:
That wasn't nearly as difficult as I thought it would be.